tests/contrib/logging/test_tracer_logging.py

```diff
@@ -0,0 +1,446 @@
+    assert status == 0, err
+    assert out == b""
+
+    if dd_trace_log_file is not None:
+
+        if dd_trace_debug == "true":
+            assert "ddtrace.commands.ddtrace_run" in str(err)  # comes from ddtrace-run debug logging
+        else:
+            assert err == b""
+
+        assert_file_contains_log(tmpdir.strpath + "/" + dd_trace_log_file)
+
+        assert out == b""
+
+    else:
+        if PY2:
+            assert 'No handlers could be found for logger "ddtrace' in err
+        else:
+            assert b"ddtrace warning log" in err
+            if dd_trace_debug == "true":
+                assert "ddtrace.commands.ddtrace_run" in str(err)  # comes from ddtrace-run debug logging
+
+
+def test_logs_with_basicConfig(run_python_code_in_subprocess, ddtrace_run_python_code_in_subprocess):
+    """
+    When DD_TRACE_DEBUG is not set and logging.basicConfig() is set, then logs are emitted
+    to stderr.
+    """
+    for run_in_subprocess in [run_python_code_in_subprocess, ddtrace_run_python_code_in_subprocess]:
+
+        code = """
+import logging
+import ddtrace
+
+logging.basicConfig(format='%(message)s')
+ddtrace_logger = logging.getLogger('ddtrace')
+
+assert ddtrace_logger.getEffectiveLevel() == logging.WARN
+assert len(ddtrace_logger.handlers) == 0
+
+ddtrace_logger.warning('warning log')
+ddtrace_logger.debug('debug log')
+"""
+
+        out, err, status, pid = run_in_subprocess(code)
+        assert status == 0, err
+        assert re.search(LOG_PATTERN, str(err)) is None
+        assert b"warning log" in err
+        assert b"debug log" not in err
+        assert out == b""
+
+
+def test_warn_logs_can_go_to_file(run_python_code_in_subprocess, ddtrace_run_python_code_in_subprocess, tmpdir):
+    """
+    When DD_TRACE_DEBUG is false and DD_TRACE_LOG_FILE_LEVEL hasn't been configured,
+    warn logs are emitted to the path defined in DD_TRACE_LOG_FILE.
+    """
+    env = os.environ.copy()
+    log_file = tmpdir.strpath + "/testlog.log"
+    env["DD_TRACE_LOG_FILE"] = log_file
+    env["DD_TRACE_FILE_SIZE_BYTES"] = "200000"
+    patch_code = """
+import logging
+import ddtrace
+
+ddtrace_logger = logging.getLogger('ddtrace')
+assert ddtrace_logger.getEffectiveLevel() == logging.WARN
+assert len(ddtrace_logger.handlers) == 1
+assert isinstance(ddtrace_logger.handlers[0], logging.handlers.RotatingFileHandler)
+assert ddtrace_logger.handlers[0].maxBytes == 200000
+assert ddtrace_logger.handlers[0].backupCount == 1
+
+ddtrace_logger.warning('warning log')
+"""
+
+    ddtrace_run_code = """
+import logging
+
+ddtrace_logger = logging.getLogger('ddtrace')
+assert ddtrace_logger.getEffectiveLevel() == logging.WARN
+assert len(ddtrace_logger.handlers) == 1
+assert isinstance(ddtrace_logger.handlers[0], logging.handlers.RotatingFileHandler)
+assert ddtrace_logger.handlers[0].maxBytes == 200000
+assert ddtrace_logger.handlers[0].backupCount == 1
+
+ddtrace_logger.warning('warning log')
+"""
+
+    for run_in_subprocess, code in [
```

Comment:
This is neat, I hadn't thought about passing multiple strings to the for loop before.